standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/MsckPartitionExpressionProxy.java

```diff
@@ -1,114 +0,0 @@
-package org.apache.hadoop.hive.metastore;
-/*
- * Licensed to the Apache Software Foundation (ASF) under one
- * or more contributor license agreements.  See the NOTICE file
- * distributed with this work for additional information
- * regarding copyright ownership.  The ASF licenses this file
- * to you under the Apache License, Version 2.0 (the
- * "License"); you may not use this file except in compliance
- * with the License.  You may obtain a copy of the License at
- * <p>
- * http://www.apache.org/licenses/LICENSE-2.0
- * <p>
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-import java.nio.charset.StandardCharsets;
-import java.util.ArrayList;
-import java.util.HashSet;
-import java.util.List;
-import java.util.Set;
-
-import org.apache.hadoop.hive.metastore.api.FieldSchema;
-import org.apache.hadoop.hive.metastore.api.FileMetadataExprType;
-import org.apache.hadoop.hive.metastore.api.MetaException;
-import org.apache.hadoop.hive.metastore.utils.FileUtils;
-import org.apache.hadoop.hive.ql.io.sarg.SearchArgument;
-import org.apache.hadoop.util.StringUtils;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
-// This is added as part of moving MSCK code from ql to standalone-metastore. There is a metastore API to drop
-// partitions by name but we cannot use it because msck typically will contain partition value (year=2014). We almost
-// never drop partition by name (year). So we need to construct expression filters, the current
-// PartitionExpressionProxy implementations (PartitionExpressionForMetastore and HCatClientHMSImpl.ExpressionBuilder)
-// all depend on ql code to build ExprNodeDesc for the partition expressions. It also depends on kryo for serializing
-// the expression objects to byte[]. For MSCK drop partition, we don't need complex expression generator. For now,
-// all we do is split the partition spec (year=2014/month=24) into filter expression year='2014' and month='24' and
-// rely on metastore database to deal with type conversions. Ideally, PartitionExpressionProxy default implementation
-// should use SearchArgument (storage-api) to construct the filter expression and not depend on ql, but the usecase
-// for msck is pretty simple and this specific implementation should suffice.
```

Comment:
I don't know - have you explored using the SA approach ? seeing the ql class thru reflection looks a bit wierd.

other way around could be to move Expr related classes out from ql (not sure if that's possible)

Reply:
Hm..hm.. using SARG will complicate a lot i guess and moving ExprNode related classes from ql is not trivial (we could explore this a bit further) but looking at a big picture most of the ExprNode classes is dependent on serde classes since we don't want serde classes in standalone-metastore We should either put class the ExprNode related class in some other module or new a new module which both ql and standalone-metastore can use.

I could think of a another approach even though it is hacky, It solves the purpose. - Since PartitionExpressionForMetastore class is required only during partition pruning step, We can switch back the expression proxy class to MsckPartitionExpressionProxy once the partition pruning step is done. This way we could solve the compatability issue.

@kgyrtkirk Any thoughts on validity of the above approach?

Reply:
or another approach could be when we fail to deserizalize the expr bytes with kyro on exception simply try converting byte array to string

Reply:
@kgyrtkirk Any thoughts on above two approaches?

Reply:
I have changed the approach, Please take a look at the new implementation.